src/execution_plan/ops/op_update.c

```diff
@@ -25,6 +25,12 @@ static int _UpdateEntity(GraphEntity *ge, PendingUpdateCtx *update) {
 	SIValue new_value = update->new_value;
 	Attribute_ID attr_id = update->attr_id;
 
+	// If this entity has been deleted, perform no updates and return early.
+	if(GraphEntity_IsDeleted(ge)) {
+		res = 0;
+		goto cleanup;
+	}
+
```

Comment:
Would it be cheaper to catch this at `_EvalEntityUpdates` somewhere around line 246, that way we won't be carrying unnecessary updates with us

Reply:
That would actually be too early; the `DeleteFree` routine needs to execute before entities are properly marked as deleted.